rewrite-core/src/main/java/org/openrewrite/TreeVisitor.java

```diff
@@ -154,6 +154,14 @@ protected void doAfterVisit(Recipe recipe) {
         if (afterVisit == null) {
             afterVisit = new ArrayList<>(2);
         }
+
+        List<Recipe> subRecipes = recipe.getRecipeList();
+        if (!subRecipes.isEmpty()) {
+            for (Recipe r : subRecipes) {
+                //noinspection unchecked
+                afterVisit.add((TreeVisitor<T, P>) r.getVisitor());
```

Comment:
Should this algorithm be recursive? And what if we see a ScanningRecipe? Throw an IllegalArgumentException?